include/eve/detail/shuffle_v2/simd/x86/shuffle_l3.hpp

```diff
@@ -63,4 +63,32 @@ requires(P::out_reg_size == P::reg_size)
   else return no_matching_shuffle;
 }
 
+template<typename P, arithmetic_scalar_value T, typename N, std::ptrdiff_t G>
+EVE_FORCEINLINE auto
+shuffle_l3_x86_blendv(P, fixed<G>, wide<T, N> x, wide<T, N> y)
+{
+  // avx512 should not be considered here at all
+  if constexpr( current_api >= avx512 ) return no_matching_shuffle;
```

Comment:
merge both if maybe ?

Reply:
really long condition that's hard to read. That's why I split

Reply:
OK then